modules/swagger-parser-v3/src/test/java/io/swagger/parser/v3/util/OpenAPIDeserializerTest.java

```diff
@@ -0,0 +1,581 @@
+
+        final OpenAPI openAPI = result.getOpenAPI();
+        Assert.assertNotNull(openAPI);
+
+        final Paths paths = openAPI.getPaths();
+        Assert.assertNotNull(paths);
+        Assert.assertEquals(paths.size(), 14);
+
+        //parameters operation get
+        PathItem petByStatusEndpoint = paths.get("/pet/findByStatus");
+        Assert.assertNotNull(petByStatusEndpoint.getGet());
+        Assert.assertNotNull(petByStatusEndpoint.getGet().getParameters());
+        Assert.assertEquals(petByStatusEndpoint.getGet().getParameters().size(), 1);
+        Assert.assertNotNull(petByStatusEndpoint.getGet().getParameters().get(0).getSchema());
+        Assert.assertEquals(petByStatusEndpoint.getGet().getParameters().get(0).getSchema().getFormat(), "int64");
+        Assert.assertEquals(petByStatusEndpoint.getGet().getParameters().get(0).getSchema().getXml().getNamespace(), "http://example.com/schema/sample");
+        Assert.assertEquals(petByStatusEndpoint.getGet().getParameters().get(0).getSchema().getXml().getPrefix(), "sample");
+        Assert.assertEquals(petByStatusEndpoint.getGet().getParameters().get(0).getIn(),"query");
+    }
+
+
+    @Test(dataProvider = "data")
+    public void readExternalDocsObject(JsonNode rootNode) throws Exception {
+        final OpenAPIDeserializer deserializer = new OpenAPIDeserializer();
+        final SwaggerParseResult result = deserializer.deserialize(rootNode);
+
+        Assert.assertNotNull(result);
+
+        final OpenAPI openAPI = result.getOpenAPI();
+        Assert.assertNotNull(openAPI);
+
+        final ExternalDocumentation externalDocumentation = openAPI.getExternalDocs();
+        Assert.assertNotNull(externalDocumentation);
+        Assert.assertNotNull(externalDocumentation.getUrl());
+        Assert.assertEquals(externalDocumentation.getUrl(),"http://swagger.io");
+
+        Assert.assertNotNull(externalDocumentation.getDescription());
+        Assert.assertEquals(externalDocumentation.getDescription(),"Find out more about Swagger");
+
+    }
+
+    @Test(dataProvider = "data")
+    public void readPathsObject(JsonNode rootNode) throws Exception {
+
+        final OpenAPIDeserializer deserializer = new OpenAPIDeserializer();
+        final SwaggerParseResult result = deserializer.deserialize(rootNode);
+
+        Assert.assertNotNull(result);
+
+        final OpenAPI openAPI = result.getOpenAPI();
+        Assert.assertNotNull(openAPI);
+
+
+        final Paths paths = openAPI.getPaths();
+        Assert.assertNotNull(paths);
+        Assert.assertEquals(paths.size(), 14);
+
+
+
+        PathItem petEndpoint = paths.get("/pet");
+        //System.out.println("$REF: "+ petEndpoint.getRef());
+        Assert.assertNotNull(petEndpoint);
+        Assert.assertEquals(petEndpoint.getSummary(),"summary");
+        Assert.assertEquals(petEndpoint.getDescription(),"description");
+        Assert.assertNotNull(petEndpoint.getPost().getExternalDocs());
```

Comment:
There is a issue that needs to be solve to do this review. Reference Object:
https://github.com/swagger-api/swagger-core/issues/2245